keepalive/keepalive.go

```diff
@@ -0,0 +1,30 @@
+package keepalive
```

Comment:
The separate package looks weird to me as well. We may want to figure out alternatives.